### PR TITLE
Fix unable to filter model correctly in case of filter options contain omitted field

### PR DESCRIPTION
**Pull request by khanhtc1202**
**What this PR does / why we need it**:

The root cause of this issue is that we have to convert model objects to `map[string]interface{}` in order to make filter logic works. The previous converting logic does: marshal object first, then unmarshal raw byte to `map[string]interface{}`.

Since our objects which are generated by protobuf are always contain `omitempty` as its json tag. While doing "first marshal", all fields with zero value will be omitted and unable to filter the object correctly if the filter options contain those omitted fields.

Through this PR, I change to using protojson marshaler instead of encoding/json marshal, which has an option to enable us to keep zero value for fields with `omitempty` jsontag.

```golang
	// remarshal entity as map[string]interface{} struct.
	m := protojson.MarshalOptions{EmitUnpopulated: true}
	raw, err := m.Marshal(pe)
```

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/datastore/filedb/filter_test.go [hunk 1]
```diff
@@ -21,42 +21,43 @@ import (
 	"github.com/stretchr/testify/require"
 
 	"github.com/pipe-cd/pipecd/pkg/datastore"
+	"github.com/pipe-cd/pipecd/pkg/model"
 )
 
 func TestConvertCamelToSnake(t *testing.T) {
```
Comment (knanao) [inline pkg/datastore/filedb/filter_test.go:27]:
```suggestion
func TestNormalizeFieldName(t *testing.T) {
```

**Review by knanao — COMMENTED** [review]
Looks neat.
/lgtm

**Comment by nghialv** [comment]
Got it. 👍 

/approve

**Pull request merged**